src/features/common/Layout/UserPropsContext.tsx

```diff
@@ -98,36 +104,66 @@ export const UserPropsProvider: FC = ({ children }) => {
     logout({ returnTo: returnUrl });
   };
 
+  const fetchUserProfile = useCallback(
+    async (impersonationData?: ImpersonationData) => {
+      const header = {
+        'tenant-key': `${tenantConfig.id}`,
+        'X-SESSION-ID': await getsessionId(),
+        Authorization: `Bearer ${token}`,
+        'x-locale': locale,
+      };
+      try {
+        const response = await fetch(
+          `${process.env.API_ENDPOINT}/app/profile`,
+          {
+            method: 'GET',
+            headers: setHeaderForImpersonation(header, impersonationData),
+          }
+        );
+
+        if (!response.ok) {
+          const errorData = await response.json();
+          throw new APIError(response.status, errorData);
+        }
+        return response.status === 204 ? true : await response.json();
+      } catch (err) {
+        console.error('Error during API call:', err);
+        throw err;
```

Comment:
The function catches errors but then immediately logs and rethrows them. This pattern means errors are logged twice - once in fetchUserProfile and again in the calling functions. Consider either logging or rethrowing, but not both.

Reply:
I removed the try/catch block from the `fetchUserProfile()` function since the caller function already has a try/catch block. If an error occurs, it will propagate to the caller function, where the error is being handled.

Reply:
I have removed the try-catch block from fetchUserProfile since the calling function handles all possible error scenarios in both places where it is used.